### ISSUE TITLE
Seeking clarification - Behaviour when mounted optional configmap appears after the pod is started

### ISSUE THREAD

**Issue by k-wall** [kubernetes/website#26611]
I am seeking clarification of the expected behaviour when a pod mounts a configmap (marking optional: true) and that configmap appears only appears *after* the pod has started.

```
      volumes:
        - name: foo
          configMap:
            name: foo
            optional: true
      containers:
      - name: controller
        volumeMounts:
        - name: foo
          mountPath: /foo
```

Will the pod see the configmap content?  Experimenting with v1.19, I believe the answer is yes, but I want to understand if this is defined behaviour and hence safe to rely on.

I don't think the docs make the point clear.
https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#understanding-configmaps-and-pods

I see the point raised a couple of times during the feature's proposal:

https://github.com/kubernetes/kubernetes/issues/37041#issue-190191613
> A config-map reference that is labeled as optional will show up as an empty directory. In addition, if the referenced config-map come into existence at a later point, it seems natural to have the config-map content show up in the mounted volume.

And during review of #175
https://github.com/kubernetes/community/pull/175#discussion_r93053986

> Define what happens if an optional configmap is missing when the pod is started on the node, and the configmap appears later while the pod is running... will the files be populated at that point, or not?

but this point wasn't answered directly.

**Comment by sftim** [comment]
Hi @k-wall 

We're happy to track an issue about the page being unclear - it's worth making it clear what happens if you add a ConfigMap where an existing Pod has this as an optional filesystem-based configuration source.

In terms of confirming the defined behavior is, that's a little outside of SIG Docs' usual purview. SIG Storage looks after ConfigMap as an API so what might make sense is if you submit a PR (pull request) that documents the behavior you've seen, and then request a review from **@kubernetes\/sig-storage-pr-reviews**.

Does that make sense?
/language en
/sig storage

**Comment by sftim** [comment]
Fixed in #26957
